src/main/java/io/cryostat/net/web/http/api/v2/TargetCredentialsGetHandler.java

```diff
@@ -0,0 +1,111 @@
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package io.cryostat.net.web.http.api.v2;
+
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.inject.Inject;
+
+import io.cryostat.configuration.CredentialsManager;
+import io.cryostat.core.log.Logger;
+import io.cryostat.net.AuthManager;
+import io.cryostat.net.security.ResourceAction;
+import io.cryostat.net.web.http.HttpMimeType;
+import io.cryostat.net.web.http.api.ApiVersion;
+import io.cryostat.platform.ServiceRef;
+
+import com.google.gson.Gson;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpMethod;
+
+class TargetCredentialsGetHandler extends AbstractV2RequestHandler<List<ServiceRef>> {
+
+    private final CredentialsManager credentialsManager;
+
+    @Inject
+    TargetCredentialsGetHandler(
+            AuthManager auth, CredentialsManager credentialsManager, Gson gson, Logger logger) {
+        super(auth, gson);
+        this.credentialsManager = credentialsManager;
+    }
+
+    @Override
+    public boolean requiresAuthentication() {
+        return true;
+    }
+
+    @Override
+    public ApiVersion apiVersion() {
+        return ApiVersion.V2_1;
+    }
+
+    @Override
+    public HttpMethod httpMethod() {
+        return HttpMethod.GET;
+    }
+
+    @Override
+    public Set<ResourceAction> resourceActions() {
+        return EnumSet.of(ResourceAction.READ_CREDENTIALS);
+    }
+
+    @Override
+    public String path() {
+        return basePath() + "credentials";
+    }
+
+    @Override
+    public HttpMimeType mimeType() {
+        return HttpMimeType.JSON;
+    }
+
+    @Override
+    public boolean isAsync() {
+        return false;
+    }
+
+    @Override
+    public IntermediateResponse<List<ServiceRef>> handle(RequestParameters requestParams)
+            throws Exception {
+        return new IntermediateResponse<List<ServiceRef>>()
+                .addHeader(HttpHeaders.CONTENT_TYPE, HttpMimeType.JSON.mime())
```

Comment:
https://github.com/cryostatio/cryostat/blob/74abc58319b8c503190edc141593aad89ae5983f/src/main/java/io/cryostat/net/web/http/api/v2/AbstractV2RequestHandler.java#L189

The `Content-Type` header is automatically set by the V2 handler abstract parent class based on the `mimeType()` method - no need to set it here